### ISSUE TITLE
minio behind nginx (terminates tls) has a 2.x minute delay and does not show all updates (using watch cmd) 

### ISSUE THREAD

**Issue by ghost** [minio/minio#3819]
## Expected Behavior

watch should show all updates and changes in the bucket

## Current Behavior

minio seems to have more than 2 minutes of delay in displaying changes to/in a bucket using the watch cmd when it is behind a nginx server working as reverse proxy. also it seems to suppress some change notifications.

minio directly connected:

[2017-03-01T09:57:04Z]  32KiB ObjectCreated http://127.0.0.1:9000/bucket/playlist.json
[2017-03-01T09:57:04Z]   382B ObjectCreated http://127.0.0.1:9000/bucket/settings-1024x768.json
[2017-03-01T09:57:04Z]   385B ObjectCreated http://127.0.0.1:9000/bucket/settings-1920x1080.json
[2017-03-01T09:57:04Z]   385B ObjectCreated http://127.0.0.1:9000/bucket/settings-1920x1200.json
[2017-03-01T09:57:04Z]  18KiB ObjectCreated http://127.0.0.1:9000/bucket/silkscreen.ttf
[2017-03-01T09:57:04Z]     0B ObjectCreated http://127.0.0.1:9000/bucket/werbung.json
[2017-03-01T10:01:28Z]  27KiB ObjectCreated http://127.0.0.1:9000/bucket/nachrichten.json
[2017-03-01T10:06:09Z]  27KiB ObjectCreated http://127.0.0.1:9000/bucket/nachrichten.json

minio connected through nginx:

[2017-03-01T09:57:04Z]  32KiB ObjectCreated https://minio.domain.tld/bucket/playlist.json
[2017-03-01T09:57:04Z]   382B ObjectCreated https://minio.domain.tld/bucket/settings-1024x768.json
[2017-03-01T09:57:04Z]   385B ObjectCreated https://minio.domain.tld/bucket/settings-1920x1080.json

## Steps to Reproduce (for bugs)

1. start minio server ./directory (just as normal)
2. setup nginx reverse proxy using the following cfg. https://gist.github.com/xstillwill/ec2dd9e6582debe7c78ea966b06eecd4
3. mirror a directory upstream 'mc mirror --watch --force --remove minio/bucket'
4. watch the bucket for changes 'mc --debug watch minio/bucket'

## Context

mirror a changing directory 1:N (server:clients) with all its content asap. (replace rsync)

content distribution to digital signage slaves

## Your Environment

* nginx version: nginx/1.10.3
* debian 8.7
* Minio version 0.0.0 (output of minio -v (using latest version May 1st, 2017))

**Comment by vadmeste** [comment]
By the way, I was able to reproduce this problem locally even with simple nginx configuration:
```
client_max_body_size 1G;

server {
 listen 8081;
 server_name localhost;
 location / {
   proxy_set_header Host $http_host;
   proxy_pass http://localhost:9000;
 }
}
```

**Comment by vadmeste** [comment]
@xstillwill, can you add `proxy_buffering  off;` under your location section of your ngnix config and test again ?

**Comment by ghost** [comment]
@vadmeste this fixed it!!! awesome! thank you so much 👍 💯

**Comment by vadmeste** [comment]
We need to test more and add that flag to the documentation.

**Comment by harshavardhana** [comment]
> We need to test more and add that flag to the documentation.

@koolhead17 @NitishT ^^

**Comment by harshavardhana** [comment]
Reopened the issue is not fixed yet.

**Comment by koolhead17** [comment]
Can be closed by https://github.com/minio/cookbook/pull/151

**Comment by harshavardhana** [comment]
Closing as fixed via minio/cookbook#151
